backend/src/services/board.service.ts

```diff
@@ -234,4 +234,37 @@ export class BoardsService {
       this.boards.push(board);
     }
   }
+
+  public removeEmptyBoards(numberOfBoards: number) {
+    // Fetches empty boards (No bot playing).
+    // Removing X number of boards that are empty.
+    const removeIndex = [];
+    this.boards.forEach((board, index) => {
+      if (Object.keys(board.getBots()).length === 0) {
+        // Do not remove board 1
+        if (board.getId() != 1) {
+          removeIndex.push(index);
+        }
+      }
+    });
+
+    // Read backwards (Removing the higher ids first)
+    removeIndex
+      .slice()
+      .reverse()
+      .forEach((removeIndex, index) => {
+        if (index < numberOfBoards) {
+          this.boards.splice(removeIndex, 1);
+        }
+      });
+    // Fetch the highest board id.
+    const nextId = Math.max.apply(
+      Math,
+      this.boards.map(function(board) {
+        return board.getId();
+      }),
+    );
+    // Set nextId to avoid gaps in the id sequence
+    Board.setNextId(nextId + 1);
```

Comment:
See my previous comment about remapping ids more dynamically on existing boards instead of setting the counter of future ids.

Reply:
:+1: